src/storage/mvcc/txn.rs

```diff
@@ -255,17 +255,42 @@ impl<S: Snapshot> MvccTxn<S> {
         };
 
         if let Some(lock) = self.reader.load_lock(&key)? {
-            // Abort on lock belonging to other transaction.
             if lock.ts != self.start_ts {
-                return Err(Error::PessimisticLockNotFound {
-                    start_ts: self.start_ts,
-                    key: key.into_raw()?,
-                });
-            }
-            if lock.lock_type != LockType::Pessimistic {
-                // Duplicated command. No need to overwrite the lock and data.
-                MVCC_DUPLICATE_CMD_COUNTER_VEC.prewrite.inc();
-                return Ok(());
+                // Abort on lock belonging to other transaction if
```

Comment:
This block is too long, extract a function is better.

Reply:
I think now it is clearer than extracting a function, and the function needs many parameters.

Reply:
Only 3 parameters: `lock`, `key`, `for_update_ts`.

Reply:
And also `is_pessimistic_lock`. The name of function is redundent,  like `check_pessimistic_prewrite_lock_conflict`.

Reply:
No, we only to extract the lines inside `if lock.ts != self.start_ts`

Reply:
OK, then we can extract the lines after `if is_pessimistic_lock ` block to a function.

Reply:
The name can be `check_non_pessimistic_lock_conflict`

Reply:
ok